Dockerfile

```diff
@@ -19,58 +20,44 @@ RUN curl https://models.silero.ai/models/tts/ru/v3_1_ru.pt -o ./silero_model.pt
 
 FROM python:3.9-slim-bullseye
 
-# Я в душе не чаю, зачем нужна следующая команда, но без неё вызов pip install на архитектуре linux/arm64 падает.
-# Связаные ссылки:
-# https://webcache.googleusercontent.com/search?q=cache:R6EG8IVigbMJ:https://pythontechworld.com/issue/pypa/pip/11435&cd=1&hl=ru&ct=clnk&gl=ru
-# О да! Решение, найденное в удалённой ветке какого-то форума!
-# https://github.com/hofstadter-io/hof/commit/838e8bbe2171ba8b9929b0ffa812c0f1ed61e975#diff-185fff912be701240e6e971b5548217a2027904efe9e365728169da65eb4983b
-RUN ln -s /bin/uname /usr/local/bin/uname
-# https://github.com/docker/buildx/issues/495#issuecomment-995503425
-RUN ln -s /usr/bin/dpkg-split /usr/sbin/dpkg-split
-RUN ln -s /usr/bin/dpkg-deb /usr/sbin/dpkg-deb
-RUN ln -s /bin/rm /usr/sbin/rm
-RUN ln -s /bin/tar /usr/sbin/tar
+ARG BUILD_USER=python
+ARG BUILD_HOME=/home/${BUILD_USER}
+ARG BUILD_DIR=irene
+ARG FULL_PATH=${BUILD_HOME}/${BUILD_DIR}
 
-RUN --mount=type=cache,target=/var/cache,sharing=locked apt update && apt install -y libportaudio2
+RUN groupadd --gid 1001 ${BUILD_USER} && useradd --create-home ${BUILD_USER} --uid 1001 --gid ${BUILD_USER} \
+&& mkdir ${FULL_PATH}
 
-RUN groupadd --gid 1001 python && useradd --create-home python --uid 1001 --gid python
-USER python:python
-WORKDIR /home/python
+COPY --chown=${BUILD_USER}:${BUILD_USER} . ${FULL_PATH}
 
-COPY ./requirements-docker.txt ./requirements.txt
-RUN --mount=type=cache,target=/home/python/.cache,uid=1001,gid=1001 pip install -r ./requirements.txt
-RUN --mount=type=cache,target=/home/python/.cache,uid=1001,gid=1001 pip install thefuzz
+RUN cp ${FULL_PATH}/runva_webapi_docker.json ${FULL_PATH}/runva_webapi.json \
+&& cp ${FULL_PATH}/docker_plugins/* ${FULL_PATH}/plugins \
+&& cp ${FULL_PATH}/plugins_inactive/plugin_tts_silero_v3.py ${FULL_PATH}/plugins
 
-#COPY * .
-COPY --chown=python:python empty_folder ./irene
-COPY lingua_franca ./irene/lingua_franca
-COPY media ./irene/media
-COPY mic_client ./irene/mic_client
-COPY model ./irene/model
-COPY mpcapi ./irene/mpcapi
-COPY plugins ./irene/plugins
-COPY utils ./irene/utils
-COPY webapi_client ./irene/webapi_client
+RUN --mount=target=/var/lib/apt/lists,type=cache,sharing=locked \
+    --mount=target=/var/cache/apt,type=cache,sharing=locked \
+    --mount=type=cache,target=/root/.cache \
+    rm -f /etc/apt/apt.conf.d/docker-clean \
+&& apt update \
+&& apt -y  -o 'Acquire::Retries=3' upgrade \
+&& apt-get install -y -o 'Acquire::Retries=3' \
+    libportaudio2 \
+&& pip install -r ./${FULL_PATH}/requirements-docker.txt \
+&& pip install thefuzz
```

Comment:
> Original Dockerfile creating lots of layers by executing multiple RUN and COPY

That did enable efficient docker cache usage, including reuse of layers between different versions.

Current version will re-run all these commands on any change (build wil get much longer) and will make users download new layers containing all of the dependencies with every new version (`docker pull` will take more time and use more internet traffic).